### ISSUE TITLE
english2rap - like translation, but translate from english to rap

### ISSUE THREAD

**Issue by silky** [silky/ideas#197]
@mobeets

**Comment by mobeets** [comment]
Haha I love this idea. How would one go about this? I can imagine something as simple as trying to substitute in rhyming synonyms, or maybe autoencoders would work too?

**Comment by silky** [comment]
here's an idea:

1. take millions of rap songs, make it learn a word embedding by predicting the next word for a given sentence
2. compare this embedding with the 'word2vec' thing
3. use this embedding to translate between them

it might be pretty bad, but maybe it's also not the worst idea

**Comment by kyluca** [comment]
This is an awesome idea! hahaha

**Comment by mobeets** [comment]
yeah that sounds reasonable. here's a good post on finding a corpus: http://dataffiti.com/2016/01/17/analyzing-rap-lyrics-part-1-of-3-creating-a-corpus/

**Comment by mobeets** [comment]
it'd also be fun to translate lines between rappers, say between tupac and biggie
